searx/engines/yummly.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Yummly
+"""
+
+from urllib.parse import urlencode
+
+from searx.utils import markdown_to_text
+
+about = {
+    "website": 'https://yummly.com',
+    "wikidata_id": 'Q8061140',
+    # not used since it requires an api key
+    "official_api_documentation": 'https://developer.yummly.com/documentation.html',
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+paging = True
+categories = []
+
+api_url = "https://mapi.yummly.com"
+number_of_results = 10
+show_pro_recipes = False
+
+
+def request(query, params):
+    args = {
+        'q': query,
+        'start': (params['pageno'] - 1) * number_of_results,
+        'maxResult': number_of_results,
+    }
+
+    params['url'] = f"{api_url}/mapi/v23/content/search?{urlencode(args)}&allowedContent=single_recipe"
+
+    return params
+
+
+def response(resp):
+    results = []
+
+    json = resp.json()
+
+    for result in json['feed']:
+        # don't show pro recipes since they can't be accessed without an account
+        if not show_pro_recipes and result['proRecipe']:
+            continue
+
+        content = result['seo']['web']['meta-tags']['description']
+        description = result['content']['description']
+        if description is not None:
+            content = markdown_to_text(description['text'])
+
+        results.append(
+            {
+                'url': result['display']['source']['sourceRecipeUrl'],
```

Comment:
Sometimes the sourceReceipeUrl seems not to work, in my tests by example I got

- `https://www.yummly.com/private/recipe/Chickpea-Kartoffelpuffer-9656329?layout=prep-steps`

I think it is better to use ..

![image](https://github.com/searxng/searxng/assets/554536/04a0b16f-6257-43c3-96ca-444431236f24)

... I will fix it.

Reply:
`sourceRecipeUrl` shows the original source of the recipe, e.g. the New York times, while the tracking ID only provides the link to the Yummly recipe. So, in order to get the real recipe source, we should use `sourceRecipeUrl or ... + tracking-id`.

Reply:
> `sourceRecipeUrl` shows the original source of the recipe, e.g. the New York times,

ah! .. thanks for pointing out, I placed a WIP commit on top of your branch, you can look at this after your vacation, now enjoy your vacation and switch off :relaxed:

Reply:
Seems to work as expected now, thanks for having a look at that!

I've been trying to rebase the changes which somewhat didn't work (because I have been acting a bit stupid), so I will by trying to fix that now.

Reply:
Sorry I missed a `/` in my WIP commit .. its now [fixed](https://github.com/searxng/searxng/compare/e16abfb815cd75997d08a242ab91734338b3a97d..25d00c52466b2aa3df0971c5fd5e479467437cd7)